modules/jwplayerTargeting.js

```diff
@@ -0,0 +1,158 @@
+import { config } from '../src/config.js';
+import { ajaxBuilder } from '../src/ajax.js';
+import { logError, isPlainObject } from '../src/utils.js';
+import { getGlobal } from '../src/prebidGlobal.js';
+import { module } from '../src/hook.js';
+
+const segCache = {};
+let requestCount = 0;
+let requestTimeout;
+let resumeBidRequest;
+
+function setup () {
+  config.getConfig('jwTargeting', (config) => {
+    // fetch media ids
+    fetchTargetingInformation(config.jwTargeting)
+  });
+
+  getGlobal().requestBids.before(onFetchCompletion);
```

Comment:
Link to doc on `requestBids.before` would be great

Reply:
it's not documented. This was suggested in the ticket https://jwplayer.atlassian.net/browse/JW8-11235 (see section `Blocking bid requests`).
It allows you to block a bid request. We use this to ensure that a request does not go out until we have finished getting the jwpsegs for the mediaIDs listed in the config.